openl3/models.py

```diff
@@ -148,6 +150,53 @@ def load_audio_embedding_model(input_repr, content_type, embedding_size, fronten
     return m
 
 
+def load_audio_embedding_model_from_path(model_path, embedding_size, frontend='kapre'):
+    """
+    Loads a model with weights at the given path. Filename is expected to follow
+    the format: `*._<input_repr>_<content_type>.h5` or
+    `*._<input_repr>_<content_type>-.*.h5`, since the model configuration will
+    be parsed from the filename.
+
+    Parameters
+    ----------
+    model_path : str
+        Path to model weights HDF5 (.h5) file. Must be in format
+        `*._<input_repr>_<content_type>.h5` or
+        `*._<input_repr>_<content_type>-.*.h5`, since model configuration
+        will be determined from the filename.
+    embedding_size : 6144 or 512
+        Embedding dimensionality.
+    frontend : "kapre" or "librosa"
+        The audio frontend to use. If frontend == 'kapre', then the kapre frontend will
+        be included. Otherwise no frontend will be added inside the keras model.
+
+    Returns
+    -------
+    model : tf.keras.Model
+        Model object.
+    """
+    model_stem = os.path.splitext(os.path.basename(model_path))[0]
+    model_spec_str = model_stem.split('-', 1)[0]
+    input_repr, content_type = model_spec_str.split('_')[-2:]
```

Comment:
I think this would be the safest:
```python
def load_audio_embedding_model_from_path(model_path, input_repr, embedding_size, frontend='kapre'):
```
but this could also be an option
```python
def load_audio_embedding_model_from_path(model_path, embedding_size, input_repr=None, frontend='kapre'):
    if not input_repr:
        model_stem = os.path.splitext(os.path.basename(model_path))[0]
```

Reply:
Yeah you're right, this is safer, the parsing isn't really necessary for this